scheduler/src/cook/kubernetes/compute_cluster.clj

```diff
@@ -1,147 +1,26 @@
-                                 nil ; labelSelector
-                                 nil ; limit
-                                 nil ; resourceVersion
-                                 nil ; timeoutSeconds
-                                 nil ; watch
-                                 )
-        node-name->node (pc/map-from-vals (fn [^V1Node node]
-                                            (-> node .getMetadata .getName))
-                                          (.getItems current-nodes))]
-    (reset! current-nodes-atom node-name->node)
-    (let [watch (WatchHelper/createNodeWatch api-client (-> current-nodes .getMetadata .getResourceVersion))]
-      (.submit kubernetes-executor ^Callable
-      (fn []
-        (try
-          (handle-watch-updates current-nodes-atom watch (fn [n] (-> n .getMetadata .getName)) nil)
-          (catch Exception e
-            (log/warn e "Error during node watch")
-            (initialize-node-watch api-client current-nodes-atom))
-          (finally
-            (.close watch))))))))
-
-(defn to-double
-  [^Quantity q]
-  (-> q .getNumber .doubleValue))
-
-(defn convert-resource-map
-  [m]
-  {:mem (if (get m "memory")
-          (-> m (get "memory") to-double (/ (* 1024 1024)))
-          0.0)
-   :cpus (if (get m "cpu")
-           (-> m (get "cpu") to-double)
-           0.0)})
-
-(defn get-capacity
-  [node-name->node]
-  (pc/map-vals (fn [^V1Node node]
-                 (-> node .getStatus .getCapacity convert-resource-map))
-               node-name->node))
-
-(defn get-consumption
-  [pod-name->pod]
-  (let [node-name->pods (group-by (fn [^V1Pod p] (-> p .getSpec .getNodeName))
-                                  (vals pod-name->pod))
-        node-name->requests (pc/map-vals (fn [pods]
-                                           (->> pods
-                                                (map (fn [^V1Pod pod]
-                                                       (let [containers (-> pod .getSpec .getContainers)
-                                                             container-requests (map (fn [^V1Container c]
-                                                                                       (-> c
-                                                                                           .getResources
-                                                                                           .getRequests
-                                                                                           convert-resource-map))
-                                                                                     containers)]
-                                                         (apply merge-with + container-requests))))
-                                                (apply merge-with +)))
-                                         node-name->pods)]
-    node-name->requests))
+           (java.util.concurrent Executors ScheduledExecutorService TimeUnit)))
 
 (defn generate-offers
```

Comment:
@pschorf  could you please fill in the data shape of what this produces?

Reply:
Done